sync/src/synchronizer/mod.rs

```diff
@@ -51,27 +52,60 @@ struct BlockFetchCMD {
     sync: Synchronizer,
     p2p_control: ServiceControl,
     recv: channel::Receiver<FetchCMD>,
+    can_start: bool,
+    number: BlockNumber,
 }
 
 impl BlockFetchCMD {
-    fn run(&self) {
+    fn run(&mut self) {
         while let Ok(cmd) = self.recv.recv() {
             match cmd {
                 FetchCMD::Fetch(peers) => {
-                    for peer in peers {
-                        if let Some(fetch) =
-                            BlockFetcher::new(&self.sync, peer, IBDState::In).fetch()
-                        {
-                            for item in fetch {
-                                BlockFetchCMD::send_getblocks(item, &self.p2p_control, peer);
+                    if self.can_start() {
+                        for peer in peers {
+                            if let Some(fetch) =
+                                BlockFetcher::new(&self.sync, peer, IBDState::In).fetch()
+                            {
+                                for item in fetch {
+                                    BlockFetchCMD::send_getblocks(item, &self.p2p_control, peer);
+                                }
                             }
                         }
+                    } else {
+                        let best_known = self.sync.shared.state().shared_best_header_ref();
+                        let number = best_known.number();
+                        if number != self.number && number % 10000 == 0 {
+                            self.number = number;
+                            info!(
+                                "best known header number: {}, difficulty: {:#x}, \
+                                 require min header number on 500_000, min difficulty: {:#x}, \
+                                 then start to download block",
+                                number,
+                                best_known.total_difficulty(),
+                                self.sync.shared.state().min_chain_work()
+                            );
+                        }
                     }
                 }
             }
         }
     }
 
+    fn can_start(&mut self) -> bool {
+        if self.can_start {
+            true
+        } else if self.sync.shared.consensus().is_mainnet {
```

Comment:
Use `min_chain_work > 0`. Checking whether the chain is mainnet can be done when setting the configurations during boot.

Reply:
This is my main reason to block the merge.